service/src/main/java/uk/gov/hmcts/reform/fpl/service/HearingBookingService.java

```diff
@@ -50,4 +52,19 @@ public HearingBooking getHearingBooking(final List<Element<HearingBooking>> hear
             .map(Element::getValue)
             .orElse(HearingBooking.builder().build());
     }
+
+    public HearingBooking getHearingBookingByUUID(List<Element<HearingBooking>> hearingBookings, UUID elementId) {
+        return hearingBookings.stream()
+            .filter(hearingBookingElement -> hearingBookingElement.getId().equals(elementId))
+            .map(Element::getValue)
+            .findFirst()
+            .orElse(HearingBooking.builder().build());
```

Comment:
Theoretically code should never hit this line as there's always `HearingBooking` for given UUID. But if not, then it should throw exception, so we actually know that something is really wrong.

And if we really don't want to throw an exception for some reason, then we should return null rather than empty `HearingBooking` - its properties are final, which means we should not be able to instantiate it with null values. `.builder().build()` allows us to do so, but that should be avoided IMO.

Reply:
By making this return `null`, although it shouldn't happen we have to ensure that the possibility of returning `null` is accounted for in each situation, therefore the use in `CaseManagementOrderService.createNextHearingDateLabel(...)` has been altered to wrap in an optional and return an empty string if the optional is empty was returned.